src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  }
+
+  if (dateMap['YY']) {
+    const yy = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = (yy < 30 ? '20' : '19') + dateMap['YY'];
+  }
+
+  const newDateParts = toFormat.map((part) => dateMap[part]);
```

Comment:
Before mapping over `toFormat`, verify that all required parts are available in `dateMap`. If a part is missing, the function may return incorrect results or fail.